include/measurement_kit/ooni/ooni_test.hpp

```diff
@@ -0,0 +1,59 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+#ifndef MEASUREMENT_KIT_OONI_TEST_HPP
+#define MEASUREMENT_KIT_OONI_TEST_HPP
+
+#include <ctime>
+#include <measurement_kit/report.hpp>
+#include <sstream>
+
+namespace mk {
+namespace ooni {
+
+class OoniTest : public NetTest, public NonCopyable, public NonMovable {
+    // Note: here we make the reasonable assumption that the owner of this
+    // instance would keep it safe until the final callback is fired
```

Comment:
Used to be `OoniTestImpl`. Moved here just the skeleton and implementation in `.cpp` file.